netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -595,8 +638,8 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
       NettyClientHandler.this.onRstStreamRead(streamId, errorCode);
     }
 
-    @Override public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data)
-        throws Http2Exception {
+    @Override
+    public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data) throws Http2Exception {
       Http2Ping p = ping;
       if (p != null) {
```

Comment:
User initiated pings cause p to be set to a new http2ping, where as the flow control pings just send a ping frame and leave this as null so this if should only pass for user pings.